doc/barefoot_dtel/Dtel-SONiC.md

```diff
@@ -0,0 +1,509 @@
+# Dataplane Telemetry in SONiC 
+# High Level Design
+Revision 0.1
+
+# About this Manual
+This document provides general information on the implementation of the Dataplane Telemetry (DTel) feature in SONiC.
+
+# Scope
+This document describes the high level design of the Dataplane Telemetry feature in SONiC.
+
+# Definitions/Abbreviation
+
+Definitions/Abbreviation  | Description
+------------------------- | -------------
+DTEL / DTel	             | Dataplane Telemetry
+ACL                       | Access Control List
+SAI                       | Switch Abstraction Interface
+INT                       | In-band Network Telemetry
+DSCP							| Differentiated Services Code Point
+OID								| Object IDentifier
+PTF			  | Packet Test Framework
+
+# Sub-system Overview 
+For more information on Dataplane Telemetry, please refer to 
+[Dataplane Telemetry SAI API](https://github.com/opencomputeproject/SAI/blob/master/doc/DTEL/SAI-Proposal-Data-Plane-Telemetry.md).
+
+The following figure depicts the overall architecture of SONiC and where Dataplane Telemetry components fit in:
+
+![Dataplane Telemetry in SONiC](Dtel_sonic.jpg)
+
+__Figure 1: Dataplane Telemetry in SONiC__. 
+
+
+Components of SONiC that will be modified or newly added are discussed in the following sub-sections.
+
+## Config DB 
```

Comment:
Thanks for the feedback! Regarding using tables in config_db, if we remove them, we run into the problem where events are re-ordered upon config replay. Since, DTel has specific dependency requirements between events, can we keep the tables until SONiC has better infrastructure to control event ordering (within an orchagent)? Another option is to create two separate orch agents, one to handle pre-requisite config and another for dependent config, and order the init of these orch agents.

Reply:
Thanks for the feedback! Regarding using tables in config_db, if we remove them, we run into the problem where events are re-ordered upon config replay. Since, DTel has specific dependency requirements between events, can we keep the tables until SONiC has better infrastructure to control event ordering (within an orchagent)?

Reply:
@liatgrozovik - i think the question is whether this feature should bear the burden with resolving an issue that exists in sonic for all features? In spirit of making progress, we can have a short term solution as we did right now and adhere to long term solution when its designed properly and reviewed by community. Clearly issue of dependency / order of operations is something to be resolved for all features.

Reply:
@liatgrozovik - @lguohan was making a few other suggestions to us on this topic, @shruthi9 can give summary on what she will implement